File: UCF HSPT Documents/2009/Solutions/jeopardy.c
#include <stdio.h>
#include <math.h>

int main ()
{
   //In this problem, we have two unknowns, lets call them A and B
   //if our two inputs are X, and Y, then we have...
   //A + B = X
   //A - B = Y
   //With a little math, we can solve for A and B, getting
   //B = (X - Y) / 2
   //A = Y + (X - Y) / 2
   //A problem is insolvable if A or B are below zero, or X - Y is not even
   //since that would result in a fraction
   
   int X = 0, Y = 0;
   FILE *fp = fopen("jeopardy.in", "r");
   
   fscanf(fp, "%d %d", &X, &Y);
   
   //Read until both inputs are zero
   while (X != -1 || Y != -1)
   {
      //Find B
      int B = (X - Y) / 2;
      
      //Find A
      int A = Y + (X - Y) / 2;
      
      //If X - Y are even and A is at least 0 and B is at least zero,
      //We have a valid solution
      if ((((abs(X - Y)) % 2) == 1)  || A < 0 || B < 0)
         printf("NOT SO FAST, TREBEKK!\n");
      else
         printf("What are the sum and difference of %d and %d?\n", A, B);
         
      fscanf(fp, "%d %d", &X, &Y);
   }
}

File: UCF HSPT Team/src/y2009/jeopardy.java
import java.io.*;
import java.util.*;

public class jeopardy {

	public static void main(String[] args) throws Exception{
		
		// Sets up file i/o and reads in the first input.
		Scanner In = new Scanner(new File("jeopardy.in"));
		int i = In.nextInt(), j = In.nextInt();
		
		// Continues running until terminating conditions are met (the -1 -1 case),
		while(i != -1 && j != -1){

			/* We know two numbers (we'll call them a & b) form the sum i and the difference j
			 * In other words, a+b=i and a-b=j. So 2a = i+j. If i+j is odd, that means that
			 * a would have to be a fraction. And if i was less than j, that means that the
			 * sum is smaller than the difference which is only possible if b is negative.
			 * Thus we print it as an invalid case.
			 */
			if((i+j)%2 == 1 || i < j)
				System.out.println("NOT SO FAST, TREBEKK!");
			else{
				// Solve for a, then use a to solve for b.
				int a = (i+j)/2, b = i-a;
				// Since the difference is non-negative, we know a will always be greater than
				// or equal to b in a valid case.
				System.out.println("What are the sum and difference of "+a+" and "+b+"?");
			}
			// Take in the next input
			i = In.nextInt();
			j = In.nextInt();
		}
	}
}
